### PR TITLE
br: add doc for auto-tune

### PR DESCRIPTION
**Pull request by ran-huang**
### What is changed, added or deleted? (Required)

add doc for auto-tune

### Which TiDB version(s) do your changes apply to? (Required)

**Tips for choosing the affected version(s):**

By default, **CHOOSE MASTER ONLY** so your changes will be applied to the next TiDB major or minor releases. If your PR involves a product feature behavior change or a compatibility change, **CHOOSE THE AFFECTED RELEASE BRANCH(ES) AND MASTER**.

For details, see [tips for choosing the affected versions](https://github.com/pingcap/docs/blob/master/CONTRIBUTING.md#guideline-for-choosing-the-affected-versions).

- [x] master (the latest development version)
- [x] v5.4 (TiDB 5.4 versions)
- [ ] v5.3 (TiDB 5.3 versions)
- [ ] v5.2 (TiDB 5.2 versions)
- [ ] v5.1 (TiDB 5.1 versions)
- [ ] v5.0 (TiDB 5.0 versions)
- [ ] v4.0 (TiDB 4.0 versions)
- [ ] v3.1 (TiDB 3.1 versions)
- [ ] v3.0 (TiDB 3.0 versions)
- [ ] v2.1 (TiDB 2.1 versions)

### What is the related PR or file link(s)?

- This PR is translated from: https://github.com/pingcap/docs-cn/pull/8081, https://github.com/pingcap/docs-cn/pull/8146, https://github.com/pingcap/docs-cn/pull/8193
- Other reference link(s):

### Do your changes match any of the following descriptions?

- [ ] Delete files
- [ ] Change aliases
- [ ] Need modification after applied to another branch 
- [ ] Might cause conflicts after applied to another branch

### CODE REVIEW

File: br/br-auto-tune.md [hunk 1]
```diff
@@ -0,0 +1,89 @@
...
+```
+|--------| The server has 8 logical CPU cores.
+|****----| By default, `backup.num-threads` is `4`. Note that auto-tune makes sure that the thread pool size is never larger than `backup.num-threads`.
+|^^****--| By default, `auto-tune-remain-threads` = round(8 * 0.2) = 2. Auto-tune adjusts the size of the thread pool to `4`.
+|^^^^**--| Because the cluster workload gets higher, auto-tune adjusts the size of the thread pool to `2`. After that, the cluster still has 2 idle CPU cores.
+```
+
+In the **Backup CPU Utilization** panel, you can see the size of the thread pool adjusted by auto-tune:
+
+![Grafana dashboard example of backup auto-tune metrics](/media/br/backup-auto-throttle.png)
```
Comment (ran-huang) [inline br/br-auto-tune.md:87]:
```suggestion
![Grafana dashboard example of backup auto-tune metrics](/media/br/br-auto-throttle.png)
```

File: br/br-auto-tune.md [hunk 2]
```diff
@@ -0,0 +1,89 @@
...
+
+# BR Auto-Tune <span class="version-mark">New in v5.4.0</span>
+
+Before TiDB v5.4.0, when you back up data using BR, the number of threads used for backup makes up 75% of the logical CPU cores. Without a speed limit, the backup process can consume a lot of cluster resources, which has a considerable impact on the cluster performance. Although you can reduce the impact of backup by adjusting the size of the thread pool, it is a tedious task to observe the CPU load and manually adjust the thread pool size.
+
+To reduce the impact of backup tasks on the cluster, starting from TiDB v5.4.0, BR introduces the auto-tune feature. When the cluster resource utilization is high, BR automatically limits the resources used by backup tasks and thereby reduces the impact on the cluster. The auto-tune feature is enabled by default.
+
+## User scenario
+
+If you hope to reduce the impact of backup tasks on the cluster, you can enable the auto-tune feature. With this feature enabled, BR performs backup tasks as fast as possible without excessively affecting the cluster.
```
Comment (en-jin19) [inline br/br-auto-tune.md:14]:
```suggestion
If you want to reduce the impact of backup tasks on the cluster, you can enable the auto-tune feature. With this feature enabled, BR performs backup tasks as fast as possible without excessively affecting the cluster.
```

File: br/br-auto-tune.md [hunk 3]
```diff
@@ -0,0 +1,89 @@
...
+
+        The backup process includes lots of SST decoding, encoding, compression, and decompression, which consume CPU resources. In addition, previous test cases have shown that during the backup process, the CPU utilization of the thread pool used for backup is close to 100%. This means that the backup tasks take up a lot of CPU resources. By adjusting the number of threads used by the backup tasks, TiKV can limit the CPU cores used by backup tasks, thus reducing the impact of backup tasks on the cluster performance.
+
+- Issue 2: For **clusters with hotspots**, backup tasks on the TiKV node that has hotspots might be excessively limited, which slows down the overall backup process.
+
+    - Solution: Eliminate the hotspot node, or disable auto-tune on the hotspot node (this might reduce the cluster performance).
+
+- Issue 3: For scenarios with **high traffic jitter**, because auto-tune adjusts the speed limit on a fixed interval (1 minute by default), it might not be able to handle high traffic jitter. For details, see [`auto-tune-refresh-interval`](#implementation).
+
+    - Solution: Diable auto-tune.
```
Comment (YuJuncen) [inline br/br-auto-tune.md:56]:
```suggestion
    - Solution: Disable auto-tune.
```

File: br/br-auto-tune.md [hunk 4]
```diff
@@ -0,0 +1,89 @@
...
+
+Alternatively, you can limit the backup speed by using the TiKV configuration item [`backup.num-threads`](/tikv-configuration-file.md#num-threads-1) or using the parameter `--ratelimit`.
+
+## Use auto-tune
+
+The auto-tune feature is enabled by default, without additional configuration.
+
+> **Note:**
+>
+> For clusters that upgrade from versions earlier than v5.4.0 to v5.4.0 or later versions, the auto-tune feature is disabled by default. You need to manually enable it.
```
Comment (YuJuncen) [inline br/br-auto-tune.md:24]: In fact, only cluster update from v5.3.0 would disable this by default.

(I just noticed that the same issue also applies to Chinese document...😢)

File: br/br-auto-tune.md [hunk 5]
```diff
@@ -0,0 +1,89 @@
...
+
+    - Auto-tune controls the resources used by the backup tasks and ensures that at least `backup.auto-tune-remain-threads` cores are available for other tasks on the same node.
+    - The default value: `round(0.2 * vCPU)`.
+    - Unit: core.
+
+- `backup.auto-tune-refresh-interval`:
+
+    - Every `backup.auto-tune-refresh-interval` minute(s), auto-tune refreshes the statistics and recalculates the maximum number of CPU cores that backup tasks can use.
+    - The default value: `1`.
+    - Unit: minute.
```
Comment (YuJuncen) [inline br/br-auto-tune.md:74]: It is a [`ReadableDuration`](https://github.com/tikv/tikv/blob/5f0e9aeb036bbf3d0cb7213fa46da7a70a75faa1/src/config.rs#L2259) and seems other documents prefer to don't mention the unit for those types.

image

**Review by YuJuncen — COMMENTED** [review]
(Maybe I need to create a PR for docs-cn for fixing those problems too...)

File: br/br-auto-tune.md (hunk 4 above)
Comment (ran-huang) [reply br/br-auto-tune.md:24]: What's the behavior of clusters < v5.3.0?
And how about clusters of v5.3.x (e.g. v5.3.1)?

**Comment by ran-huang** [comment]
@YuJuncen @en-jin19 Aligned with the new Chinese PR. PTAL again. Thanks!

**Review by en-jin19 — APPROVED** [review]
LGTM

**Review by YuJuncen — APPROVED** [review]
LGTM

**Pull request merged**